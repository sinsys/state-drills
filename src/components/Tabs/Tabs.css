.Tabs-wrapper {
	max-width: 400px;
}