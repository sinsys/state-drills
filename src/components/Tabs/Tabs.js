import React from 'react';
import './Tabs.css';

class Tabs extends React.Component {

	static defaultProps = {
		tabs: []
	};

	constructor(props) {
		super(props);
		this.state = {
			currentTabIndex: 0
		};
	};

	renderButtons() {
		return (
			this.props.tabs.map((tab, index) => (
			<button
				key={index}
				onClick={() => this.handleButtonClick(index)}
			>
				{tab.name}
			</button>
		)));
	};

	renderContent() {
		const currentTab = this.props.tabs[this.state.currentTabIndex];
		return (
			<div className="content">
				{currentTab.content}
			</div>
		)
	};

	handleButtonClick = (index) => {
		this.setState({
			currentTabIndex: index
		})
	};

	render() {
		return (
			<div className="Tabs-wrapper">
				<h2>Tabs</h2>
				{this.renderButtons()}
				{!!this.props.tabs.length && this.renderContent()}
			</div>
		)
	}
}

export default Tabs;