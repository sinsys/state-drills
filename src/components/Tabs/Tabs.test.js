import React from 'react';
import ReactDOM from 'react-dom';
// import renderer from 'react-test-renderer';
import Tabs from './Tabs';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe(`Tabs Component`, () => {
	const tabsProp = 
	  [
	    { name: 'First tab',
	      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque.' },
	    { name: 'Second tab',
	      content: 'Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit.' },
	    { name: 'Third tab',
	      content: 'Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur.' }
	  ];

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Tabs />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // JEST + react-test-renderer
  // it('renders the first tab by default', () => {
  // 	const tree = renderer.create(
  // 		<Tabs tabs={tabsProp} />
  // 	)
  // 	expect(tree).toMatchSnapshot()
  // });

  // ENZYME
  it('renders the first tab by default', () => {
  	const wrapper = shallow(<Tabs />);
  	expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('renders empty when given no tabs', () => {
  	const wrapper = shallow(<Tabs />);
  	expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('closes the first tab and opens the clicked tab', () => {
  	const wrapper = shallow(<Tabs tabs={tabsProp} />);
		// console.log('>>> WRAPPER <<<')
		// console.log(wrapper.debug())
		// console.log('>>> FIND(BUTTON) <<<')
		// console.log(wrapper.find('button').at(2).debug())
		wrapper.find('button').at(2).simulate('click');
		expect(toJson(wrapper)).toMatchSnapshot();
  })
})