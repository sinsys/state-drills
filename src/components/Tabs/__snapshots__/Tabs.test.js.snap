// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component closes the first tab and opens the clicked tab 1`] = `
<div
  className="Tabs-wrapper"
>
  <h2>
    Tabs
  </h2>
  <button
    key="0"
    onClick={[Function]}
  >
    First tab
  </button>
  <button
    key="1"
    onClick={[Function]}
  >
    Second tab
  </button>
  <button
    key="2"
    onClick={[Function]}
  >
    Third tab
  </button>
  <div
    className="content"
  >
    Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur.
  </div>
</div>
`;

exports[`Tabs Component renders empty when given no tabs 1`] = `
<div
  className="Tabs-wrapper"
>
  <h2>
    Tabs
  </h2>
</div>
`;

exports[`Tabs Component renders the first tab by default 1`] = `
<div
  className="Tabs-wrapper"
>
  <h2>
    Tabs
  </h2>
</div>
`;
